internal/tofu/node_provider.go

```diff
@@ -113,10 +120,21 @@ func (n *NodeApplyableProvider) ConfigureProvider(ctx EvalContext, provider prov
 	}
 
 	configSchema := resp.Provider.Block
-	configVal, configBody, evalDiags := ctx.EvaluateBlock(configBody, configSchema, nil, EvalDataForNoInstanceKey)
-	diags = diags.Append(evalDiags)
-	if evalDiags.HasErrors() {
-		return diags
+
+	var configVal cty.Value
+	var evalDiags tfdiags.Diagnostics
+
+	// Check if the config body is of type TestProviderConfig.
+	// Use the value directly if available
+	testProviderConfig, isTestProviderConfig := configBody.(*configs.TestProviderConfig)
+	if isTestProviderConfig && testProviderConfig.Value != cty.NilVal {
+		configVal = testProviderConfig.Value
+	} else {
+		configVal, configBody, evalDiags = ctx.EvaluateBlock(configBody, configSchema, nil, EvalDataForNoInstanceKey)
+		diags = diags.Append(evalDiags)
```

Comment:
this diag should probably be wrapped in a more user-friendly one. similar to below.

Reply:
added [here.](https://github.com/opentofu/opentofu/pull/1347/commits/b15cb4962966b088eebc00b34e6dac0470d81eaa)